core/src/main/java/org/apache/calcite/jdbc/Driver.java

```diff
@@ -52,7 +52,7 @@
 public class Driver extends UnregisteredDriver {
   public static final String CONNECT_STRING_PREFIX = "jdbc:calcite:";
 
-  final Function0<CalcitePrepare> prepareFactory;
+  public final Function0<CalcitePrepare> prepareFactory;
```

Comment:
This should remain package-private; but change the type to `Supplier`. Add a method

```
public CalcitePrepare createPrepare() {
  return prepareFactory.get();
}
```